### PR TITLE
Validate the cluster field of job config

### PR DESCRIPTION
**Pull request by hongkailiu**
/cc @alvaroaleman @stevekuznetsov

### CODE REVIEW

File: prow/cmd/checkconfig/main.go [hunk 1]
```diff
@@ -1059,3 +1065,30 @@ func validateTideContextPolicy(cfg *config.Config) error {
 
 	return utilerrors.NewAggregate(errs)
 }
+
+func validateClusterField(cfg *config.Config) error {
+	var errs []error
+	for orgRepo, jobs := range cfg.PresubmitsStatic {
+		for _, job := range jobs {
+			if job.Cluster != "" && job.Agent != "kubernetes" && job.Agent != "" {
```
Comment (stevekuznetsov) [inline prow/cmd/checkconfig/main.go:1073]: Can we DRY this out?

```go
func validateCluster(base config.JobBase) error {
    if base.Cluster != "" && (job.Agent == "" || job.Agent == prowapi.KubernetesAgent) {
        return fmt.Errorf("%s: cannot set cluster field if agent is not 'kubernetes' or empty", base.name)
    }
    return nil
}

func validateClusterField(cfg *config.Config) error {
	var errs []error
	for orgRepo, jobs := range cfg.PresubmitsStatic {
		for _, job := range jobs {
			if err := validateCluster(job.Base); err != nil {
				errs = append(errs, fmt.Errorf("%s: %w", orgRepo, err))
			}
		}
	}
	// ....
}
```

Comment (hongkailiu) [reply prow/cmd/checkconfig/main.go]: Updated.

File: prow/cmd/checkconfig/main.go [hunk 2]
```diff
@@ -1059,3 +1065,35 @@ func validateTideContextPolicy(cfg *config.Config) error {
 
 	return utilerrors.NewAggregate(errs)
 }
+
+func validateClusterFieldForJobBase(job config.JobBase) error {
```
Comment (petr-muller) [inline prow/cmd/checkconfig/main.go:1069]: `ForJobBase` stutters (in both signature and callsite). How about calling this one `validateJobCluster`?

File: prow/cmd/checkconfig/main.go [hunk 3]
```diff
@@ -1059,3 +1065,35 @@ func validateTideContextPolicy(cfg *config.Config) error {
...
+	if job.Cluster != "" && !(job.Agent == "" || job.Agent == string(v1.KubernetesAgent)) {
+		return fmt.Errorf("%s: cannot set cluster field if agent is not 'kubernetes' or empty", job.Name)
+	}
+	return nil
+}
+
+func validateClusterField(cfg *config.Config) error {
```
Comment (petr-muller) [inline prow/cmd/checkconfig/main.go:1076]: just `validateCluster`?

File: prow/cmd/checkconfig/main.go (hunk 2 above)
Comment (hongkailiu) [reply prow/cmd/checkconfig/main.go]: Updated.

File: prow/cmd/checkconfig/main.go (hunk 3 above)
Comment (hongkailiu) [reply prow/cmd/checkconfig/main.go]: Done.

**Review by petr-muller — APPROVED** [review]

File: prow/cmd/checkconfig/main.go [hunk 4]
```diff
@@ -1059,3 +1065,35 @@ func validateTideContextPolicy(cfg *config.Config) error {
 
 	return utilerrors.NewAggregate(errs)
 }
+
+func validateJobCluster(job config.JobBase) error {
+	if job.Cluster != "" && !(job.Agent == "" || job.Agent == string(v1.KubernetesAgent)) {
```
Comment (alvaroaleman) [inline prow/cmd/checkconfig/main.go:1070]: That would be a lot easier to read as `(job.Agent != "" || job.Agent != string(v1.KubernetesAgent)`. Also, shouldn't this be opt in for the agents that do not support it? There is e.G. knative, no idea if it supports setting this field or ppl might write their own agent.

Comment (hongkailiu) [reply prow/cmd/checkconfig/main.go]: TIL `agent: knative`.

I have not found any of it in test-infra jobs and (red hat) release jobs through.
If `knative` does not support `cluster`, we can added it into the set later.

**Review by alvaroaleman — APPROVED** [review]

**Pull request merged**